### PR TITLE
SWC-2274

### PR DESCRIPTION
**Pull request by kimyen**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/entity/renderer/WikiSubpagesWidget.java [hunk 1]
```diff
@@ -187,9 +192,13 @@ public void onFailure(Throwable caught) {
...
 	}
-	
+
 	public interface UpdateOrderHintCallback {
 		void updateOrderHint(List<String> newOrderHintIdList);
 	}
-	
+
+	// for test only
+	public void setEditOrderButtonVisible(){
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/renderer/WikiSubpagesWidget.java]: remove

File: src/test/java/org/sagebionetworks/web/unitclient/widget/entity/renderer/WikiSubpagesTest.java [hunk 2]
```diff
@@ -150,6 +157,30 @@ public void testAsWidget() {
 		widget.asWidget();
 		verify(mockView).asWidget();
 	}
+
+	@Test
+	public void testEditOrderButtonVisibilityForAnonymous(){
+		when(mockAuthenticationController.isLoggedIn()).thenReturn(false);
+		widget.configure(new WikiPageKey(entityId, ObjectType.ENTITY.toString(), null), descriptor, null, null, null, false);
```
Comment (kimyen) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/entity/renderer/WikiSubpagesTest.java:164]: Wiki place

File: src/test/java/org/sagebionetworks/web/unitclient/widget/entity/renderer/WikiSubpagesTest.java [hunk 3]
```diff
@@ -150,6 +157,30 @@ public void testAsWidget() {
 		widget.asWidget();
 		verify(mockView).asWidget();
 	}
+
+	@Test
+	public void testEditOrderButtonVisibilityForAnonymous(){
+		when(mockAuthenticationController.isLoggedIn()).thenReturn(false);
+		widget.configure(new WikiPageKey(entityId, ObjectType.ENTITY.toString(), null), descriptor, null, null, null, false);
+		verify(mockView).setEditOrderButtonVisible(false);
+		widget.configure(new WikiPageKey(entityId, ObjectType.ENTITY.toString(), null), descriptor, null, null, null, true);
```
Comment (kimyen) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/entity/renderer/WikiSubpagesTest.java:166]: Synapse place

File: src/test/java/org/sagebionetworks/web/unitclient/widget/entity/renderer/WikiSubpagesTest.java [hunk 4]
```diff
@@ -150,6 +157,30 @@ public void testAsWidget() {
...
 	}
+
+	@Test
+	public void testEditOrderButtonVisibilityForAnonymous(){
+		when(mockAuthenticationController.isLoggedIn()).thenReturn(false);
+		widget.configure(new WikiPageKey(entityId, ObjectType.ENTITY.toString(), null), descriptor, null, null, null, false);
+		verify(mockView).setEditOrderButtonVisible(false);
+		widget.configure(new WikiPageKey(entityId, ObjectType.ENTITY.toString(), null), descriptor, null, null, null, true);
+		verify(mockView, Mockito.times(2)).setEditOrderButtonVisible(false);
+		AsyncMockStubber.callFailureWith(new Throwable()).when(mockSynapseClient).getV2WikiOrderHint(any(WikiPageKey.class), any(AsyncCallback.class));
```
Comment (kimyen) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/entity/renderer/WikiSubpagesTest.java:168]: fails to get order hint

**Pull request merged**